### ISSUE TITLE
java.net.SocketTimeoutException causes client to lock up and freeze completely

### ISSUE THREAD

**Issue by hjdarnel** [Equinox-/runelite-influxdb#5]
```
2020-05-28 00:16:50 [pool-3-thread-1] ERROR n.m.runelite.influxdb.InfluxDbPlugin - Failed to write to influxDB 1 times
java.lang.RuntimeException: java.net.SocketTimeoutException: timeout
	at org.influxdb.impl.InfluxDBImpl.execute(InfluxDBImpl.java:390)
	at org.influxdb.impl.InfluxDBImpl.write(InfluxDBImpl.java:260)
	at net.machpi.runelite.influxdb.write.InfluxWriter$ThrottledWriter.lambda$flush$0(InfluxWriter.java:157)
	at java.util.Optional.ifPresent(Optional.java:159)
	at net.machpi.runelite.influxdb.write.InfluxWriter$ThrottledWriter.flush(InfluxWriter.java:154)
	at net.machpi.runelite.influxdb.write.InfluxWriter$Writer.flush(InfluxWriter.java:113)
	at net.machpi.runelite.influxdb.write.InfluxWriter.lambda$flush$1(InfluxWriter.java:86)
	at java.util.concurrent.ConcurrentHashMap.forEach(ConcurrentHashMap.java:1597)
	at net.machpi.runelite.influxdb.write.InfluxWriter.flush(InfluxWriter.java:86)
	at net.machpi.runelite.influxdb.InfluxDbPlugin.flush(InfluxDbPlugin.java:146)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.net.SocketTimeoutException: timeout
	at okio.Okio$4.newTimeoutException(Okio.java:227)
	at okio.AsyncTimeout.exit(AsyncTimeout.java:284)
	at okio.AsyncTimeout$2.read(AsyncTimeout.java:240)
	at okio.RealBufferedSource.indexOf(RealBufferedSource.java:344)
	at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:216)
	at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:210)
	at okhttp3.internal.http1.Http1Codec.readResponseHeaders(Http1Codec.java:189)
	at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:75)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:120)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
	at org.influxdb.impl.GzipRequestInterceptor.intercept(GzipRequestInterceptor.java:42)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
	at okhttp3.logging.HttpLoggingInterceptor.intercept(HttpLoggingInterceptor.java:145)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:185)
	at okhttp3.RealCall.execute(RealCall.java:69)
	at retrofit2.OkHttpCall.execute(OkHttpCall.java:174)
	at org.influxdb.impl.InfluxDBImpl.execute(InfluxDBImpl.java:382)
	... 16 common frames omitted
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
	at java.net.SocketInputStream.read(SocketInputStream.java:171)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at okio.Okio$2.read(Okio.java:138)
	at okio.AsyncTimeout$2.read(AsyncTimeout.java:236)
	... 43 common frames omitted
```

I've run into this log a couple of times now. When it happens, the entire client thread locks up, so if I'm in combat or something all the damage stacks up or mechanics are missed, etc. I don't know if it's a configuration issue, influx issue, or elsewise, but an exception like this shouldn't lock the client up. I've had to disable the plugin until we can get this fixed up! Thanks for your work!

**Comment by Equinox-** [comment]
Pretty sure this is a connectivity issue on your end, but it should be
possible for me to ensure the game's thread pool doesn't get blocked on
this.  Failing that I'll just put it on a dedicated thread.

**Comment by Equinox-** [comment]
Should be fixed by 7df87cbc27f2e309453133329d8161cc8cefbee8
